### ISSUE TITLE
SortingGroupModel3D and Children

### ISSUE THREAD

**Issue by gregorsGit** [helix-toolkit/helix-toolkit#724]
Hi,
I try your new SortingGroupModel3D.
In your example it is working.
But I need a hirarchy of models.
So I add SortingGroupModel3D to the Children. Like this.
```
        public ObservableElement3DCollection TransparentGroupModelSource { get; } = new ObservableElement3DCollection();
        public SortingGroupModel3D TransparentComp { get; } = new SortingGroupModel3D();
        public SortingGroupModel3D TransparentComp2 { get; } = new SortingGroupModel3D();
...
 TransparentGroupModelSource.Add(TransparentComp);
 TransparentGroupModelSource.Add(TransparentComp2);
```
Each Child-SortingGroup is working.
The Root-SortingGroup not.
Or is there an other way to build a tree.
Thanks.

**Comment by gregorsGit** [comment]
This would be nice.
I try to change the SortingGroupNode, but with poor result.
After removing/adding childs they lost the transforms of the parent groupnode.


[SortingGroupNode.zip](https://github.com/helix-toolkit/helix-toolkit/files/1931999/SortingGroupNode.zip)

**Comment by holance** [comment]
After the sorting, your implementation move all nodes N levels below current node into the current node's items. This is incorrect and causes wrong scene graph. Cross node sorting is hard to implement.
Try the #725 order independent transparency rendering.

**Comment by gregorsGit** [comment]
Working with this version I found a strange bug.
Removing and adding GroupElements changes the visiblity appearance.

To Show what I mean I resort the model-list by camera-reset in your OrderIndependantTransparentRendering example.
Bevor pressing the button I can zoom around an see all PlaneGeometry models.

![grafik](https://user-images.githubusercontent.com/34306200/39131960-00660930-4711-11e8-88c4-5220297b4b52.png)

After resort, you must zoom closer and some PlaneGeometry´s are missing.
![grafik](https://user-images.githubusercontent.com/34306200/39131878-ceef8a8e-4710-11e8-917f-376d5186dcb8.png)

Here are my changes in the viewmodel.
[MainViewModel.zip](https://github.com/helix-toolkit/helix-toolkit/files/1938567/MainViewModel.zip)

Hope you can reproduce the bug.

**Comment by holance** [comment]
You need to make sure to mark MeshGeometryModel3D.IsTransparent = true so the model will be used for special transparency rendering

**Comment by gregorsGit** [comment]
I check that IsTransparent = true is set correct.
But no changes. Some elements disappeare changing the camera-view.

Everything is fine until I remove and add some ModelGroups.
I remove models from the list, to set them invisible and reduce memory.
This happens not so often.

Is there anything I can do after resort to reset/invalidate the view.
Or update the render-view ?

**Comment by holance** [comment]
You are adding 3 models in AddPlaneGroup in your view model, but only the last model is set IsTransparent=true. The other two are not set. I am not getting your problem once the other two models are set correctly.

**Comment by gregorsGit** [comment]
Changing all 3 models to IsTransparent=true do not help.
It seems like the frustum is changed by resorting.
It happens even when all MeshGeometryModel3D.IsTransparent=false.
The same example with the source version from 17. April. do not show this effect.

**Comment by holance** [comment]
Try to set Viewport3dx.enableviewfrustum =false

**Comment by gregorsGit** [comment]
Ok. 
My elements now all back.
Even when I rotate/zoom.
All fine.

-----------------


By testing this version I get sometimes a new mouse bug.
It happens during load / without mouse interaction.

namespace HelixToolkit.Wpf.SharpDX Viewport3DX
in MouseDownHitTest
the hits are null
and crashes. 
Cannot reproduce yet.

  var hits = this.FindHits(pt);
            //   if (hits.Count > 0)
// changed to 
            if (hits != null && hits.Count > 0)

**Comment by holance** [comment]
Fixed in #736. Close this issue for now. Please create new issue if you found anything else.
